test/typescript/providers/test.ts

```diff
@@ -7,11 +7,11 @@ describe("full integration test", () => {
       // This is a workaround for that, by having a new driver every run we don't have this bug.
       const deployDriver = new TestDriver(__dirname);
       await deployDriver.setupTypescriptProject();
-      const deployLog = deployDriver.deploy(["using-all-providers"]);
+      const deployLog = await deployDriver.deploy(["using-all-providers"]);
       expect(deployLog).toContain("null_resource.test (test) will be created");
       expect(deployLog).toContain("1 to add");
       expect(deployLog).toContain("Apply complete!");
-    });
+    }, 600_000);
```

Comment:
puh, we should take a look why it's taking so long

Reply:
Jon recently added the AWSCC provider (because that one is using `nested_type`) maybe that increased it over the limit of 300_000. A lower value might work as well, but I think 10 minutes is short enough for our tests in case something fails silently and causes a timeout.
What do you think? 400_000 might also work for example.

Reply:
I'm fine with the timeout, I was just making a mental note to explore how we can improve the test runtime here

Reply:
We could parallelize it somewhat by having different test suites for the different providers 😁